modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/exceptions/BadRequestException.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * you may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.mdm.mobileservices.windows.common.exceptions;
+
+import org.wso2.carbon.mdm.mobileservices.windows.common.util.Message;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+/**
+ * Exception class for bad request failures
+ */
+public class BadRequestException extends WebApplicationException {
+
+    public BadRequestException(Message message, MediaType mediaType) {
+        super(Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(message).
+                type(mediaType).build());
```

Comment:
If the exception type is going to represent a bad request from the client side, why we are setting response status as INTERNAL_SERVER_ERROR. Shouldn't this be Response.Status.BAD_REQUEST